mitodl/smoot-design
# chore: Add release action

### What are the relevant tickets?
For https://github.com/mitodl/hq/issues/6039

### Description (What does it do?)
This PR adds [semantic-release](https://semantic-release.gitbook.io/semantic-release) to release this package. It:
- inspects commits since last release to construct changelog +  decide next version number (major/minor/patch)
    - only commits using conventional commit are included
    - Since we usually squash commits, this repo is set to use PR title as default squashed commit message, and we enforce conventional commit PR titles.
- publishes the release to NPM and github releases page

### How can this be tested?
1. Read the [readme](https://github.com/mitodl/smoot-design/blob/e862c05eeef83eb996124c485c96e99ac9a6cefd/README.md). It should make sense.
2. View the "Release" section of https://github.com/mitodl/smoot-design/actions/runs/11846407719/job/33013978096 to see example changelog based on this branch + some empty test commits.
    - you can trigger this with github CLI via `gh workflow run release --ref cc/releasing`. Once this PR is merged, it can be manually triggered via action button.
3. Change this PR title to something that is not a conventional commit; the PR check should fail with a somewhat useful message.
    - change it back, and the message should be deleted. 

### Checklist:
- [ ] Write a README section
- [ ] remove `cc/releasing` from `.releaserc`
